### ISSUE TITLE
App router on PPR stop responding after navigate to path with query

### ISSUE THREAD

**Issue by zamarawka** [vercel/next.js#69437]
### Link to the code that reproduces this issue

https://github.com/zamarawka/reproduction-app-router-bug

### To Reproduce

1. Build app for production `npm run build`
2. Run app `npm run start`
3. Go to `localhost:3000/blog`
4. Try navigate via menu to "Tab: other" (it has path `/?tab=other`)
5. Navigation doesn't happen (clicks on `Link`s don't do anything) and router stop responding on any requests

### Current vs. Expected behavior

I expect navigation work, pages changes, router responds to `Link` clicks

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: 1.22.10
  pnpm: N/A
Relevant Packages:
  next: 15.0.0-canary.133 // There is a newer canary version (15.0.0-canary.134) available, please upgrade! 
  eslint-config-next: N/A
  react: 19.0.0-rc-f90a6bcc-20240827
  react-dom: 19.0.0-rc-f90a6bcc-20240827
  typescript: 5.3.3
Next.js Config:
  output: standalone
```

### Which area(s) are affected? (Select all that apply)

Navigation, Partial Prerendering (PPR)

### Which stage(s) are affected? (Select all that apply)

next build (local), next start (local)

### Additional context

This case doesn't happen on dev mode, only after build and run app. Order of steps to reproduce is important, if you initially going to same page without query params (e.g. `/`) and start to navigate to query (`/?tag=other`) - everything works as expected.

App become broken after update from `15.0.0-canary.112` to latest canary version, probably any version next to `112` have such behaviour. Not sure, but it looks like https://github.com/vercel/next.js/pull/68340 relates.

Also found out that you can run from browser console:

```js
window.next.router.refresh();
```

And app continue working, page updates, router start respond.

It tooks me a lot of time to figure out what going wrong and how to reproduce it. Looks like problem relates to dynamic (server-rendered on demand) pages with query params. I ended up my debugging with something going wrong in 
[navigateReducer](https://github.com/vercel/next.js/blob/9720d0adc0b0687267e35804b6f6380fa351ab0c/packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts#L108).

**Comment by Alexandredc** [comment]
Same problem here. But I don't use PPR.

**Comment by nicreaskdjasd** [comment]
@ztanner, Thank you for your fixes.I have tested with canary-136,the problem still exists for normal push, when not using PPR

**Comment by zamarawka** [comment]
> @ztanner, Thank you for your fixes.I have tested with canary-136,the problem still exists for normal push, when not using PPR

It seems like you need to open new issue.
